app/src/main/java/com/nmakademija/nmaakademija/StartActivity.java

```diff
@@ -15,12 +45,109 @@ protected void onCreate(Bundle savedInstanceState) {
         boolean notificationsEnabled = NMAPreferences.getNotifications(this);
 
         AppEvent.getInstance(this).setNotificationEnabledUserProperty(notificationsEnabled);
+
+        setContentView(R.layout.activity_start);
+
+        errorTV = (TextView) findViewById(R.id.error);
+
+        AppEvent.getInstance(this).trackCurrentScreen(this, "open_sign_in");
+
+        gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestEmail()
+                .build();
+
+        mGoogleApiClient = new GoogleApiClient.Builder(this)
+                .enableAutoManage(this, new GoogleApiClient.OnConnectionFailedListener() {
+                    @Override
+                    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+
+                    }
+                })
+                .addApi(Auth.GOOGLE_SIGN_IN_API, gso)
+                .build();
+
+        findViewById(R.id.google_sign_in_button).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                signInWithGoogle();
+            }
+        });
+
+        mCallbackManager = CallbackManager.Factory.create();
+        final LoginButton loginButton = (LoginButton) findViewById(R.id.facebook_login_button);
+        loginButton.setReadPermissions("email", "public_profile");
+        loginButton.registerCallback(mCallbackManager, new FacebookCallback<LoginResult>() {
+            @Override
+            public void onSuccess(LoginResult loginResult) {
+                if (!loginResult.getAccessToken().isExpired())
+                    signIn(FacebookAuthProvider.getCredential(loginResult.getAccessToken().getToken()));
+            }
+
+            @Override
+            public void onCancel() {
+            }
+
+            @Override
+            public void onError(FacebookException error) {
+            }
+        });
+        findViewById(R.id.anonymous_login).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startNextActivity();
```

Comment:
Why not signing in user anonymously https://firebase.google.com/docs/auth/android/anonymous-auth ?